synapse/third_party_rules/access_rules.py

```diff
@@ -291,17 +296,36 @@ def check_event_allowed(
 
         return True
 
-    def _on_rules_change(
-        self, event: EventBase, state_events: StateMap[EventBase],
+    async def check_room_can_be_added_to_public_rooms_directory(
+        self, room_id: str, state_events: StateMap[EventBase],
     ) -> bool:
-        """Implement the checks and behaviour specified on allowing or forbidding a new
-        im.vector.room.access_rules event.
+        """Implements
+        synapse.events.ThirdPartyEventRules.check_room_can_be_added_to_public_rooms_directory
+
+        Determines whether a room can be published to the public rooms directory. A room
+        with access rule "unrestricted" may not be published to the room directory.
 
         Args:
-            event: The event to check.
+            room_id: The ID of the room.
             state_events: A dict mapping (event type, state key) to state event.
-                State events in the room before the event was sent.
+                State events in the room.
 
+        Returns:
+            Whether the room is allowed to be published to the public rooms directory.
+        """
+        # We need to know the rule to apply when processing the event types below.
+        rule = self._get_rule_from_state(state_events)
+        return rule != AccessRules.UNRESTRICTED
```

Comment:
```suggestion
        return rule == AccessRules.RESTRICTED
```

Reply:
I would write like this, because here only Restricted rooms are allowed to be published

Reply:
Oh, didn't realise that's what we wanted, but sounds good.